[MIG] web_easy_switch_company: migrate to 9.0

njeudy opened the pull request:
- Use new UserMenu include method
- Works with Community and enterprise

Commit:
fix: migrate to 9.0

- Use new UserMenu include method
- Works with Community and enterprise

Commit:
fix: update README.md for migration state

njeudy commented:
@rvalyi @pedrobaeza @lmignon  this one can not be used on entreprise too ?

pedrobaeza commented:
Yeah, it's the same issue. Any module depending on web module will have the same problem. Anyway, have you tried if this module appears good with the enterprise UI?

njeudy commented:
@pedrobaeza yes it work on community and entreprise. I can post screenshot of the result if needed.

njeudy commented:
@pedrobaeza need I squash on one commit ?

pedrobaeza commented:
Well, it's not mandatory in this case, but you should know that the README.md task update is done automatically via an script that checks every module and its installability.

legalsylvain commented:
Hi @njeudy. Thanks for porting this module.
- I just tested on runbot, and it doesn't seem to work.
  (http://3135224-260-e31796.runbot2.odoo-community.org/) Do you know why ?
- Could you create one demo company, and add admin user to the right groups (demo data) (to make this module more easy to test) ?

Kind regards.

njeudy commented:
@legalsylvain: as I know you can only test on runbot when it will be merged ?

I look to you runbot instance and it is not le module I propose in this merge request.
- To test, juste add crm app and enable multi-company settings. I can just add a small description on how to test ?

njeudy commented:
@pedrobaeza ok did know that README.md is autoamticcaly updated :)

legalsylvain commented:
- @njeudy :
  no, runbot use PR code. you can see your changes [here](http://3135224-260-e31796.runbot2.odoo-community.org/web?db=3135224-260-e31796-all#id=117&view_type=form&model=ir.module.module&menu_id=51&action=36).
- It's better (*) if the module can work just installing the module with demo data. It's interesting to see all the groups needed to make the module works. (but it's "nice to have" and can be done in another PR).
  (better for promotion, test, ...)

njeudy commented:
Ok, 

To test user should have at least two allowed company in his user form. I think I can add small docs ?

I update administrator account on the runbot instance you indicate ( @legalsylvain )
And it works.

njeudy commented:
![screenshot](https://www.dropbox.com/s/3hfgcnelhy8uld1/Capture%20d%27%C3%A9cran%202015-11-19%2021.19.49.png?dl=1)

legalsylvain commented:
:+1: Thanks. (code review & test).

njeudy commented:
need to improve something on this module or is it ok ?

pedrobaeza commented on README.md:
```diff
@@ -30,7 +30,7 @@ addon | version | summary
 [web_dialog_size](web_dialog_size/) | 8.0.0.1.0 (unported) | A module that lets the user expand a dialog box to the full screen width.
 [web_dom_model_classes](web_dom_model_classes/) | 8.0.1.0.0 (unported) | Allows small UI changes with simple CSS
-[web_easy_switch_company](web_easy_switch_company/) | 8.0.1.0.0 (unported) | Multicompany - Easy Switch Company
+[web_easy_switch_company](web_easy_switch_company/) | 9.0.1.0.0 | Multicompany - Easy Switch Company
```
This is not in the correct section, so better to leave it without touching, and let the script to update this README.

pedrobaeza commented on web_easy_switch_company/__openerp__.py:
```diff
@@ -22,7 +22,7 @@
+    'version': '9.0.1.0.0',
     'category': 'web',
     'description': """
 Add menu to allow user to switch to another company more easily
```
Move this description to a README.rst file and use the OCA's latest template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

pedrobaeza replied:
Now that you have the README, please remove this key

pedrobaeza commented:
Please add a disclaimer in the README.rst file indicating that this module is not compatible with web enterprise addon.

njeudy commented:
@pedrobaeza thanks for your review ... all should be fixed now :)

sbidoul commented on web_easy_switch_company/README.rst:
```diff
@@ -0,0 +1,90 @@
+
+This module extend web backend and allow user to switch to another company more easyly.
+
+Note: Due to licence incompatibility, you have not the legal right to install this module on enterprise version ...
```
According to https://github.com/odoo/odoo/issues/9973#issuecomment-163199596 this sentence is not necessary.

eLBati replied:
I agree

njeudy commented:
@sbidoul so ? I remove it ? what is OCA advise on this ?

sbidoul commented:
@njeudy I _personally_ think you can remove it.

This topic is on the agenda of the next OCA board meeting on January 12th so we should have an official OCA position shortly after.

pedrobaeza commented:
@njeudy sorry for the noise. It seems that finally it's correct that this module can be used by both community and enterprise (at least this seems so according https://github.com/odoo/odoo/issues/9973#issuecomment-163199596), so no need for the disclaimer.

Commit:
new: Add correct README.txt and rebase:

    - Rebase to latest 9.0 branch
    - Modify README without legal claim
    - Test an check on latest odoo community and enterprise 9.0.

njeudy commented:
@pedrobaeza @sbidoul : Ok I rebase on top of V9 and update README.rst without legal claim.

Think it's pretty good now.

llacroix commented:
Any idea why this module isn't yet merged.. ?

llacroix commented on web_easy_switch_company/static/src/js/switch_company.js:
```diff
@@ -17,18 +18,27 @@
+    var Widget = require('web.Widget');
+    var UserMenu = require('web.UserMenu');
+    var SystrayMenu = require('web.SystrayMenu');
+    var web_client = require('web.web_client');
```
This line isn't necessary and the UserMenu probably too isn't necessary as you're using SystrayMenu

Commit:
fix: remove unneeded dependency

njeudy commented:
@llacroix thanks ! update done

max3903 commented on web_easy_switch_company/README.rst:
```diff
@@ -0,0 +1,88 @@
+web_easy_switch_company
+=======================
+
+This module extend web backend and allow user to switch to another company more easyly.
```
This module extends web backend and allows user to switch to another company more easily.

max3903 commented on web_easy_switch_company/README.rst:
```diff
@@ -0,0 +1,88 @@
+To configure this module, you need to:
+
+* Enable multi company in your odoo instance
+* Add your user in at least 2 company, and you will see the switch at the upper right corner
```
2 companies

max3903 commented on web_easy_switch_company/README.rst:
```diff
@@ -0,0 +1,88 @@
+Functionality:
+--------------
+    * Add a new menu in the top bar to switch to another company more easily;
+    * Remove the old behaviour to switch company;
```
No space before the star

max3903 commented:
Minor typos. Once they are fixed along with @pedrobaeza's comments, :+1:

Commit:
fix: typo on README

Commit:
fix: remove description in __openerp__.py

Commit:
fix: rebase on 9.0 and remove README.md changes

njeudy commented:
Ok I rebase on top of last 9.0 oca/web revision and update to correct all comment.

njeudy commented:
@pedrobaeza @max3903 @llacroix any chance to merge it ?

:+1:

max3903 commented:
:+1:

pedrobaeza commented:
:+1:
